File: BookStore-BE/src/main/java/com/java/bookstore/entities/CartItemEntity.java
package com.java.bookstore.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "cart_item")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CartItemEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Min(value = 0)
	private int quantity;

	private double totalPrice;
	
	@ManyToOne
	@JoinColumn(name = "book_id")
	private BookEntity book;
	
	@ManyToOne
	@JoinColumn(name = "shopping_cart_id")
	private ShoppingCartEntity shoppingCart;
}

File: BookStore-NestJS/src/entities/cart-item/entities/cart-item.entity.ts
import { Book } from "src/entities/books/entities/book.entity";
import { ShoppingCart } from "src/entities/shopping-cart/entities/shopping-cart.entity";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";

@Entity()
export class CartItem {
    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(() => ShoppingCart, shoppingCart => shoppingCart.cartItems)
    shoppingCart: ShoppingCart;

    @ManyToOne(() => Book)
    book: Book;

    @Column()
    quantity: number;

    @Column('float', { precision: 10, scale: 2 })
    totalPrice: number;
}
